### PR TITLE
Fix: detect mutating keywords inside CTEs in read-only mode

### LINKED ISSUE #271: Execute update in CTE in postgres read-only
> ❯ update contract 1 to set the site post code to 'SW11'
> 
> ⏺ The database connection is read-only — I'm unable to execute UPDATE statements through this tool.
> 
>   You'd need to run that directly against the database. The SQL would be:
> 
>   UPDATE contracts SET site_location_postcode = 'SW11' WHERE id = 1;
> 
> ❯ let's make sure it's read-only. try selecting from a CTE that updates
> 
> ⏺ db - Execute SQL (postgres) (MCP)(sql: "WITH updated AS (\n  UPDATE contracts SET site_location_postcode = 'SW11' WHERE id = 1 RETURNING id, site_location_postcode\n)\nSELECT * FROM updated")
>   ⎿  {
>        "success": true,
>        "data": {
>      … +10 lines (ctrl+o to expand)
> 
> ⏺ The database is NOT read-only — the update succeeded. The "read-only" label in the tool description is incorrect; it actually allows data-modifying statements.

Perhaps it's not possible to strictly enforce read-only here, and the documentation should be updated accordingly.

### PR DESCRIPTION
**Pull request by tianzhou**
## Summary
- Read-only SQL validation only checked the first keyword, allowing `WITH cte AS (UPDATE/INSERT/DELETE ...) SELECT ...` to bypass restrictions
- Now scans the full statement (after stripping comments and string literals) for DML/DDL keywords in WITH/SELECT/EXPLAIN contexts
- Dialect-aware REPLACE detection: matches `REPLACE INTO` statement form for MySQL/MariaDB/SQLite without false-positiving on `REPLACE()` function calls or identifiers

## Changes
- `src/utils/allowed-keywords.ts`:
  - Added mutating keyword scan for WITH statements (INSERT, UPDATE, DELETE, DROP, ALTER, CREATE, TRUNCATE, MERGE, GRANT, REVOKE, RENAME)
  - Added `REPLACE INTO` detection for MySQL/MariaDB/SQLite via per-dialect pattern map
  - Added `SELECT ... INTO` detection for both SELECT and WITH statements
  - Added `EXPLAIN ANALYZE` validation — recursively checks the inner statement
  - Handles `EXPLAIN (ANALYZE false/off)` and `EXPLAIN ANALYZE VERBOSE` correctly
  - Removed standalone `ANALYZE` from allowed keywords (it updates statistics)
  - Extracted `checkReadOnly()` to avoid re-running `stripCommentsAndStrings` on recursive calls
- `src/utils/__tests__/allowed-keywords.test.ts`: Added 40+ test cases covering CTE bypass, REPLACE handling, SELECT INTO, EXPLAIN ANALYZE, and edge cases

Closes #271

## Test plan
- [x] All 56 allowed-keywords unit tests pass
- [x] Pure SELECT CTEs still allowed
- [x] REPLACE() function calls not falsely rejected
- [x] Mutating keywords in string literals/comments not falsely rejected
- [x] No regressions in other unit tests

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/utils/allowed-keywords.ts [hunk 1]
```diff
@@ -7,35 +7,115 @@ import { stripCommentsAndStrings } from "./sql-parser.js";
...
+  "i",
+);
+
+/**
+ * Extended pattern for dialects that support REPLACE INTO (MySQL/MariaDB/SQLite).
+ * Only matches `REPLACE INTO` (with optional LOW_PRIORITY/DELAYED), not
+ * REPLACE() function calls or identifiers named `replace`.
+ */
+const mutatingPatternWithReplace = new RegExp(
+  `\\b(?:${mutatingKeywords.join("|")}|replace\\s+(?:(?:low_priority|delayed)\\s+)?into)\\b`,
```
[the remark this answers is not in the document]
Comment (tianzhou) [reply src/utils/allowed-keywords.ts]: Fixed in 87f75fe. Changed pattern to `replace(?!\s*\()` to catch all REPLACE statement forms (including `REPLACE tbl SET ...` without INTO). Accepts the minor false positive on identifiers named `replace` in MySQL/MariaDB/SQLite as a security tradeoff.

File: src/utils/__tests__/allowed-keywords.test.ts [hunk 2]
```diff
@@ -66,6 +66,158 @@ describe("isReadOnlySQL", () => {
...
+
+    it("should allow a CTE named 'replace' in MySQL", () => {
+      const sql = "WITH replace AS (SELECT 1) SELECT * FROM replace";
+      expect(isReadOnlySQL(sql, "mysql")).toBe(true);
+    });
+
+    it("should reject REPLACE (non-function) inside WITH in MySQL", () => {
+      const sql = "WITH cte AS (SELECT 1) REPLACE INTO users VALUES (1, 'test')";
+      expect(isReadOnlySQL(sql, "mysql")).toBe(false);
+    });
```
[the remark this answers is not in the document]
Comment (tianzhou) [reply src/utils/__tests__/allowed-keywords.test.ts:159]: Added in 87f75fe. Added tests for `REPLACE tbl SET ...` (both standalone and inside WITH).
